### PR TITLE
user profile page membership table

### PR DESCRIPTION
**Pull request by hlminh2000**
**Description of changes**

- Adds interpretation of user token to display program access levels.
- Also consolidates the uikit's `useTheme` hook so they all comes with flow type

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: client/components/pages/user/ProgramAccessBox/index.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        role = PROGRAM_USER_ROLES_DISPLAY.SUBMITTER;
+        permission = PROGRAM_USER_PERMISSIONS_DISPLAY.SUBMITTER;
+      } else if (canReadProgramData({ egoJwt, programId: shortName })) {
+        role = PROGRAM_USER_ROLES_DISPLAY.COLLABORATOR;
+        permission = PROGRAM_USER_PERMISSIONS_DISPLAY.COLLABORATOR;
+      }
+      return {
+        shortName,
+        role,
+        permissions: 'Submit Data, Download Data, Manage Users',
```
Comment (hlminh2000) [inline client/components/pages/user/ProgramAccessBox/index.js]: damn it!!! I missed this...

File: client/global/utils/egoJwt/index.js [hunk 2]
```diff
@@ -67,8 +67,34 @@ export const canReadProgram = (args: { egoJwt: ?string, programId: string }): bo
...
 
-export const isProgramAdmin = (args: { egoJwt: ?string, programId: string }): boolean =>
+export const isProgramAdmin: (args: { egoJwt: ?string, programId: string }) => boolean = args =>
   !!args.egoJwt && TokenUtils.isProgramAdmin(args);
 
 export const getReadableProgramShortNames = (egoJwt: string): Array<string> =>
   TokenUtils.getReadableProgramShortNames(egoJwt);
+
+export const canReadProgramData: (args: { egoJwt: ?string, programId: string }) => boolean = args =>
+  !!args.egoJwt && TokenUtils.canReadProgramData(args);
```
Comment (joneubank) [inline client/global/utils/egoJwt/index.js]: Shouldn't TokenUtils just do the check for egoJwt missing? Feels bad to have to empty property check every time....

File: client/global/utils/egoJwt/index.js [hunk 3]
```diff
@@ -67,8 +67,34 @@ export const canReadProgram = (args: { egoJwt: ?string, programId: string }): bo
...
+}) => boolean = args => !!args.egoJwt && TokenUtils.canWriteProgramData(args);
+
+export const canReadSomeProgramData: (egoJwt: ?string) => boolean = egoJwt =>
+  !!egoJwt && TokenUtils.canReadSomeProgramData(egoJwt);
+
+export const canWriteSomeProgramData: (egoJwt: ?string) => boolean = egoJwt =>
+  !!egoJwt && TokenUtils.canWriteSomeProgramData(egoJwt);
+
+export const getReadableProgramDataScopes: (egoJwt: ?string) => PermissionScopeObj[] = egoJwt =>
+  !egoJwt ? [] : TokenUtils.getReadableProgramDataScopes(egoJwt);
```
Comment (joneubank) [inline client/global/utils/egoJwt/index.js]: Same comment, this logic could be implemented in TokenUtils

File: client/uikit/Icon/index.js [hunk 4]
```diff
@@ -61,7 +65,7 @@ const Icon = ({
           ))
         ) : (
           <path
-            fill={theme.colors[fill] || fill || svg.defaultFill}
+            fill={(fill && theme.colors[fill]) || fill || svg.defaultFill}
```
Comment (joneubank) [inline client/uikit/Icon/index.js]: This path doesn't have all the same fill options as the previous... is this intentional or an oversight? If it should be consistent then move the fill resolution into a reusable method.

File: client/components/pages/user/ProgramAccessBox/index.js (hunk 1 above)
Comment (joneubank) [inline client/components/pages/user/ProgramAccessBox/index.js]: Is this meant to be hardcoded?

Comment (joneubank) [reply client/components/pages/user/ProgramAccessBox/index.js]: It is not, you've updated this.

Comment (hlminh2000) [reply client/components/pages/user/ProgramAccessBox/index.js]: just fixed this :P was a mistake

File: client/components/pages/user/ProgramAccessBox/index.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    };
+  });
+};
+
+export default function ProgramAccessBox() {
+  const { token } = useAuthContext() || {};
+  const { data, loading: loadingPrograms, error } = useQuery(USER_PROGRAM_LIST_QUERY);
+  let programs: T_ProgramTableProgram[];
+  const isDcc = isDccMember(token);
+  if (isDcc) {
```
Comment (joneubank) [inline client/components/pages/user/ProgramAccessBox/index.js]: 👍

File: client/global/utils/egoJwt/index.js (hunk 2 above)
Comment (hlminh2000) [reply client/global/utils/egoJwt/index.js]: yes it should! will update in future version :)

File: client/uikit/Icon/index.js (hunk 4 above)
Comment (hlminh2000) [reply client/uikit/Icon/index.js]: addressed! :)

**Pull request merged**